### ISSUE TITLE
Qwen Student and teacher logits mismatch during KD, needs padding

### ISSUE THREAD

**Issue by nouranali** [pytorch/torchtune#2598]
In this exact line, in kd_loss.py
![Image](https://github.com/user-attachments/assets/4255c8ac-7aca-41a1-b31e-5d9a044ba8d6)

**Comment by ebsmothers** [comment]
Hi @nouranali thanks for creating the issue! I left a comment on #2599, happy to continue the discussion there.
